client/src/models/MezzuriteDataProvider.ts

```diff
@@ -0,0 +1,49 @@
+import { TreeDataProvider, TreeItem, EventEmitter, Event } from 'vscode';
+
+import MezzuriteComponent from './MezzuriteComponent';
+import MezzuriteTreeItem from './MezzuriteTreeItem';
+
+class MezzuriteDataProvider implements TreeDataProvider<TreeItem> {
+  private _onDidChangeTreeData: EventEmitter<TreeItem> = new EventEmitter<TreeItem>();
+  readonly onDidChangeTreeData: Event<TreeItem> = this._onDidChangeTreeData.event;
+
+  constructor (
+    private components: MezzuriteComponent[],
+    private rootPath: string
+  ) {}
+
+  getChildren (element: MezzuriteTreeItem): Thenable<TreeItem[]> {
```

Comment:
Discussed offline. I think inline comments would be very helpful in explaining why the function is how it is.

Reply:
Per the offline discussion, given the new direction of moving to javascript, the comparator was abstracted into a separate testable function in the future.